src/main/java/org/eclipse/tractusx/ssi/lib/proof/types/jws/JWSProofVerifier.java

```diff
@@ -77,25 +74,38 @@ public class JWSProofVerifier implements IVerifier {
 
   private final DidResolver didResolver;
 
+  /**
+   * Verify Verifiable Document
+   *
+   * @param hashedLinkedData the hashed linked data
+   * @param document the document
+   * @return the boolean
+   * @throws SignatureParseException the signature parse exception
+   * @throws DidParseException the did parse exception
+   * @throws InvalidPublicKeyFormatException the invalid public key format exception
+   * @throws SignatureVerificationFailedException the signature verification failed exception
+   * @throws UnsupportedSignatureTypeException the unsupported signature type exception
+   */
   @SneakyThrows({DidResolverException.class})
   public boolean verify(HashedLinkedData hashedLinkedData, Verifiable document)
       throws SignatureParseException,
           DidParseException,
           InvalidPublicKeyFormatException,
-          SignatureVerificationException {
+          SignatureVerificationFailedException,
+          UnsupportedSignatureTypeException,
+          NoVerificationKeyFoundException {
 
     final Proof proof =
-        document.getProof().orElseThrow(() -> new SignatureParseException("no proof found"));
-    if (!proof.getType().equals(JWSSignature2020.JWS_VERIFICATION_KEY_2020)) {
```

Comment:
if this is removed, does that mean, that Ed25519Signature2020 is obsolete, and we do not have to expect this type anymore?

Reply:
Fixed with https://github.com/Cofinity-X/SSI-agent-lib/pull/49/commits/54b99a4ca55d4deb7c3d58799b8de51bc33c7e01